tests/hypothesis_strategies.py

```diff
@@ -0,0 +1,158 @@
+still exists a larger but finite number for the upper interval end.
+"""
+
+_decorrelate = st.one_of(
+    st.booleans(),
+    st.floats(min_value=0.0, max_value=1.0, exclude_min=True, exclude_max=True),
+)
+"""A strategy that generates decorrelation settings."""
+
+parameter_name = st.text(min_size=1)
+"""A strategy that generates parameter names."""
+
+categories = st.lists(st.text(min_size=1), min_size=2, unique=True)
+"""A strategy that generates parameter categories."""
+
+
+@st.composite
+def smiles(draw: st.DrawFn):
+    """Generate short SMILES strings."""
+    n_atoms = draw(st.integers(min_value=0, max_value=19))
+    string = "C"
+    for _ in range(n_atoms):
+        next_atom = random.choice("CNO") if string[-1] == "C" else random.choice("C")
+        string += next_atom
+    return string
+
+
+@st.composite
+def substance_data(draw: st.DrawFn):
+    """Generate data for class:`baybe.parameters.substance.SubstanceParameter`."""
+    names = draw(st.lists(st.text(min_size=1), min_size=2, max_size=10, unique=True))
+    substances = draw(st.lists(smiles(), min_size=len(names), max_size=len(names)))
+    return dict(zip(names, substances))
+
+
+@st.composite
+def custom_encodings(draw: st.DrawFn):
+    """Generate data for class:`baybe.parameters.custom.CustomDiscreteParameter`."""
+    index = st.lists(st.text(min_size=1), min_size=2, max_size=10, unique=True)
+    cols = columns(
+        names_or_number=10,
+        elements=st.floats(allow_nan=False, allow_infinity=False),
+        unique=True,
+        dtype=DTypeFloatNumpy,
+    )
+    return draw(data_frames(index=index, columns=cols))
+
+
+@st.composite
+def numerical_discrete_parameter(
+    draw: st.DrawFn,
+):
+    """Generate class:`baybe.parameters.numerical.NumericalDiscreteParameter`."""
+    name = draw(parameter_name)
+    values = draw(
+        st.lists(
+            st.one_of(
+                st.integers(),
+                st.floats(allow_infinity=False, allow_nan=False),
+            ),
+            min_size=2,
+            unique=True,
+        )
+    )
+
+    # Reject examples where the tolerance validator cannot be satisfied
+    try:
+        return NumericalDiscreteParameter(name=name, values=values)
+    except NumericalUnderflowError:
+        assume(False)
+
+
+@st.composite
+def numerical_continuous_parameter(draw: st.DrawFn):
+    """Generate class:`baybe.parameters.numerical.NumericalContinuousParameter`."""
+    name = draw(parameter_name)
+    lower = draw(st.floats(max_value=_largest_lower_interval, allow_infinity=False))
+    upper = draw(st.floats(min_value=lower, exclude_min=True, allow_infinity=False))
+    return NumericalContinuousParameter(name=name, bounds=(lower, upper))
+
+
+@st.composite
+def categorical_parameter(draw: st.DrawFn):
+    """Generate class:`baybe.parameters.categorical.CategoricalParameter`."""
+    name = draw(parameter_name)
+    values = draw(categories)
+    encoding = draw(st.sampled_from(CategoricalEncoding))
```

Comment:
For the strategies I would say "no". Because their only task is to span the space of possible objects, which is already done with current version. Adding the string option only makes the strategy unnecessarily complicated (there is also the `shrinkage` concept of hypothesis that you need to keep in mind) without changing the output at all.

We could debate about whether we tests like this could be added as additional validation/initialization tests. However, I not sure if this particular already crosses the line of what still belongs to our framework and what is outside of it. In a sense, your suggested test would test the attrs converter, not our parameter object. If you follow that logic to the end, it would mean that we would need to add tests for all input types that our converters accept. That is, if we have `converter=list`, your suggestion would translate into having explicit test cases for all object types that can be converter into lists, i.e. basically everything that implements the `typing.Sequence` interface.

**TL;DR:** I consider the particular enum case we are talking about as an edge case. If you think it should be included, we could add outside hypothesis, but I would not add it the the strategy.

Reply:
sure since the attrs converter was implemented correctly it would simpy test the list conversion which doesnt seem very exciting. But making exactly sure that it was implemented/included correctly is the point of also testing the string based initialization, who knows what will change in the attrs related lines in the future. Especially for the enums imo thats important as most suers will prob prefer the string based initialization.

im missing both contexts here what is shrinkage in the context of this package (couldnt find it in their docs either) and what the hypothesis strategies are intended to be if youre saying that wouldnt belong here. But if it requires something complicated imo its minor enough to skip it

Reply:
Ok, I see your point. Even though chances are really really small that this will break in the future, adding a short test case doesn't hurt either, so I've added them. For now, I went via two additional explicit tests instead of extending the strategy, as described above. We can still discuss in our meeting for what we intend to use hypothesis and for what not, and we can change it after the meeting if necessary. For now, I went with the credo "only span the space of possibilities". Please resolve if this is OK